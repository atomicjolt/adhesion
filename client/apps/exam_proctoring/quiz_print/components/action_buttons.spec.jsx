import React from 'react';
import { shallow } from 'enzyme';
import { hashHistory } from 'react-router';
import Buttons from './action_buttons';

describe('Action Buttons', () => {
  let result;

  beforeEach(() => {
    result = shallow(<Buttons />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the history push button', () => {
    hashHistory.push = jest.fn();
    result.find('button').first().simulate('click');
    expect(hashHistory.push).toBeCalled();
  });
});
