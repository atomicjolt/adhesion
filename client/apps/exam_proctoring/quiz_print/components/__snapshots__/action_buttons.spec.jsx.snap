// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Buttons matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <actionButtons />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="pure-button"
          onClick={[Function]}
>
          Back
</button>,
        <button
          className="pure-button button-primary"
          onClick={[Function]}
          style={
                    Object {
                              "float": "right",
                            }
          }
>
          Print
</button>,
      ],
      "className": "no-print",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Back",
          "className": "pure-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Back",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Print",
          "className": "pure-button button-primary",
          "onClick": [Function],
          "style": Object {
            "float": "right",
          },
        },
        "ref": null,
        "rendered": "Print",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="pure-button"
            onClick={[Function]}
>
            Back
</button>,
          <button
            className="pure-button button-primary"
            onClick={[Function]}
            style={
                        Object {
                                    "float": "right",
                                  }
            }
>
            Print
</button>,
        ],
        "className": "no-print",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Back",
            "className": "pure-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Back",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Print",
            "className": "pure-button button-primary",
            "onClick": [Function],
            "style": Object {
              "float": "right",
            },
          },
          "ref": null,
          "rendered": "Print",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
